src/store/plugins/account.js

```diff
@@ -55,9 +56,14 @@ export default store =>
             nameFee: txObject.nameFee && +aettosToAe(txObject.nameFee),
           },
         };
-        const fee = modal
-          ? aeToAettos(await dispatch('modals/open', confirmProps, { root: true }))
-          : txObject.fee;
+        const fee =
+          modal &&
+          (await checkPermissions({
+            method: 'transaction.sign',
+            params: { txObject: { params: { amount: txObject.amount } } },
+          }))
+            ? aeToAettos(await dispatch('modals/open', confirmProps, { root: true }))
```

Comment:
Won't it be showing two modals one by one? Is it intended behaviour?

Reply:
`checkPermissions` function doesn't show any modals, it's just one more check